internal/scheduleringester/dbops.go

```diff
@@ -254,6 +303,14 @@ func (a MarkJobsCancelled) CanBeAppliedBefore(b DbOperation) bool {
 	return !definesJob(a, b)
 }
 
+func (a UpdateJobSchedulingInfo) CanBeAppliedBefore(b DbOperation) bool {
+	return !definesJob(a, b)
+}
+
+func (a UpdateJobQueuedState) CanBeAppliedBefore(b DbOperation) bool {
```

Comment:
this feels a bit wrong in that moving a `jobrequeued` event before the job failure (or even the job lease) seems incorrect.  In practice I think this will work because of our versioning- but it might be worth speaking to Albin to see how we could tighten this up.

Reply:
actually I just had a better look at the code- I think what you've done here is fine and the real issue is with the ingester in that we don't perform all updates for a given batch inside a transaction.  I'll make a separate PR for that.

Reply:
If we put everything inside a transaction then I think it will work as expected. I think what is happening right now is that the state after all operations have been inserted is correct, but intermediate states are not correct.  If we move everything into a single transaction then this will prevent readers from seeing these intermediate states and we should be fine.  Note that this issue applies to various operations, not just the new new one you've added here.

The alternative would be to add some more complex logic around how operations may be re-ordered.  I'm not a big fan of this because the current mechanism doesn't give us enough flexibility to be able to easily represent this.  E.g. right now we can easily prevent this operation being moved backwards, but we can't easily prevent it being moved forwards (i.e. other operations being moved backwards, passing this one).  Changing this would be something of an effort and I think not necessary, as long as we ensure the final database state is correct (which we can if we use a single transaction).

tagging @severinson here in case he has an opinion.